src/framework/components/anim/controller.js

```diff
@@ -269,6 +276,36 @@ Object.defineProperties(AnimController.prototype, {
             return this._getActiveStateProgressForTime(this._timeInState);
         }
     },
+    activeStateDuration: {
+        get: function () {
+            if (this.activeStateName === ANIM_STATE_START || this.activeStateName === ANIM_STATE_END)
+                return 0.0;
+
+            var maxDuration = 0.0;
+            for (var i = 0; i < this.activeState.animations.length; i++) {
+                var activeClip = this._animEvaluator.findClip(this.activeState.animations[i].name);
+                if (activeClip && activeClip.track.duration > maxDuration) {
```

Comment:
Small point, might be neater to go:
```javascript
maxDuration = math.Max(maxDuration, activeClip.track.duration);
```

Also, ```this.activeState.animations``` is referenced multiple times, might be worth storing it in a var.

Reply:
Cheers, updated these 👍